packages/pluggableWidgets/shrinking-header-web/src/components/ShrinkingHeaderLinear.tsx

```diff
@@ -0,0 +1,65 @@
+import { createElement, CSSProperties, ReactElement, ReactNode, useCallback, useEffect, useState } from "react";
+import classNames from "classnames";
+
+import "../ui/ShrinkingHeader.scss";
+import { useWrappingDivHeight } from "../utils/WrappingDivStyler";
+
+export interface ShrinkingHeaderLinearProps {
+    rootElementRef?: (node: HTMLElement | null) => void;
+    name?: string;
+    className?: string;
+    style?: CSSProperties;
+    tabIndex?: number;
+    content?: ReactNode;
+    initHeight: number;
+    shrunkHeight: number;
+}
+
+export function ShrinkingHeaderLinear(props: ShrinkingHeaderLinearProps): ReactElement {
+    const { rootElementRef, name, className, style, tabIndex, content, initHeight, shrunkHeight } = props;
+
+    const [headerHeight, setHeaderHeight] = useState<number>();
+    const [headerElement, setHeaderElement] = useState<HTMLElement>();
+
+    const actualClassName = classNames("widget-shrinking-header", "widget-shrinking-header-linear", className);
+
+    const updateElement = useCallback(
+        (node: HTMLElement | null) => {
+            setHeaderElement(node ?? undefined);
+            rootElementRef?.(node);
+        },
+        [rootElementRef, setHeaderElement]
+    );
+
+    useEffect(() => {
+        const updateHeaderHeight = function (): void {
+            let headerHeight;
+
+            if (window.scrollY > initHeight - shrunkHeight) {
+                headerHeight = initHeight - (initHeight - shrunkHeight);
+            } else if (window.scrollY < 0) {
+                headerHeight = initHeight;
+            } else {
+                headerHeight = initHeight - window.scrollY;
+            }
+
+            setHeaderHeight(headerHeight);
+        };
```

Comment:
Is there a reason you did not throttle this one compared to the ShrinkingHeaderThreshold one?

Reply:
Linear should act immediately, otherwise the animation isn't smooth. Throttling the threshold improved performance on mobile devices. In the future, we can provide a property so that users can configure it.